sdk/apis/tenancy/v1alpha1/types_workspace.go

```diff
@@ -155,11 +155,18 @@ type WorkspaceSpec struct {
 	// +kubebuilder:validation:XValidation:rule="!has(oldSelf.path) || !has(self.path) || self.path == oldSelf.path",message="path is immutable"
 	Type WorkspaceTypeReference `json:"type,omitempty"`
 
+	// MountName is the name of the mount that this workspace is mounted under.
+	//
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="!has(oldSelf.name) || !has(self.name) || self.name == oldSelf.name",message="name is immutable"
+	// +kubebuilder:validation:XValidation:rule="!has(oldSelf.namespace) || !has(self.namespace) || self.namespace == oldSelf.namespace",message="namespace is immutable"
+	Mount *metav1.ObjectMeta `json:"mount,omitempty"`
```

Comment:
Doesn't this need some higher-level CEL rule to make it mutual exclusive and bound to type?